provider/pkg/metadata/annotations.go

```diff
@@ -35,6 +35,8 @@ const (
 	AnnotationPatchForce        = AnnotationPrefix + "patchForce"
 	AnnotationPatchFieldManager = AnnotationPrefix + "patchFieldManager"
 
+	AnnotationDeletionPropagation = AnnotationPrefix + "deletionPropagation"
```

Comment:
What about `deleteCascade` to better match the k8s terminology?

Reply:
My rationale was to match the terminology in [apimachinery](https://github.com/kubernetes/apimachinery/blob/1da46c3f5a5b4a0cc756cb6050df0cf6f06b64c2/pkg/apis/meta/v1/types.go#L479-L495) but you're right that `kubectl` uses `--cascade`. I like your suggestion too and will change it unless others have stronger opinions.

```
// DeletionPropagation decides if a deletion will propagate to the dependents of
// the object, and how the garbage collector will handle the propagation.
type DeletionPropagation string
```

Reply:
I am strongly aligned with using "propagationPolicy," which is the terminology used in the API server and embraced by other GitOps tools such as [ArgoCD](https://argo-cd.readthedocs.io/en/stable/user-guide/sync-options/#resources-prune-deletion-propagation-policy) and [ConfigSync](https://cloud.google.com/anthos-config-management/docs/how-to/managing-objects#deletion_propagation_options).

Furthermore, "orphan" isn't really a cascading deletion type, reinforcing the preference for "propagationPolicy." When kubectl introduced the `cascade` flag, its purpose was to toggle between orphan (`cascade=false`) and background cascading delete (`cascade=true`), as indicated in this upstream issue: https://github.com/kubernetes/kubernetes/issues/93364. I suggest adopting `propagationPolicy` to also accommodate potential non-cascading strategies in future Kubernetes versions.

Reply:
OK let's go with `deletionPropagationPolicy`.